hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -1389,4 +1398,8 @@ public MessageDigest getMessageDigestInstance() {
     return E_TAG_PROVIDER.get();
   }
 
+  private String extractPartsCount(String eTag) {
+    String[] parts = eTag.replace("\"", "").split("-");
+    return  parts[parts.length - 1];
```

Comment:
We can only return the `PartsCount` when the following condition is true (i.e. if the last part is a numerical value), otherwise we return `null`:
```
parts[parts.length - 1].matches("\\d+")
```